### DISCUSSION TITLE
Upload Progress with Chunked Upload

### DISCUSSION THREAD

**Discussion by OscarEstudilloMarin** [rpldy/react-uploady#714] [Q&A] [answered]
Hello,

After some issues uploading large files with regular upload, I decided to try Chunked Uploading to avoid server timeouts and other improvements.

With regular upload I was able to show a nice and smooth progress bar, however I find this impossible with chunk uploads, I only get a progress update whenever a chunk is fully uploaded.
 
I guess this can be solved including an event such as CHUNK_PROGRESS or making the ITEM_PROGRESS event consider also the chunk progress in global. 

Maybe this is difficult due to possible parallelism, in my case I upload every chunk sequentially.

Is there any possibility to get an smooth progress using chunk upload?

Thank you very much,
Oscar.

Progress with chunk upload:
https://github.com/user-attachments/assets/2d0a50be-cdb6-40c7-bfd6-d7b2bad8df67

Progress with regular upload:
https://github.com/user-attachments/assets/08bd2610-b68e-46ce-82f1-7aaf7a1f2248

**Comment by yoavniran** [comment, marked as answer]
hey @OscarEstudilloMarin,
thanks for the feedback. Definitely makes sense. 
Cool upload UI, btw. :) 

I will need to think about this and see what would be the best way.

Making item-progress incorporate chunk-progress would be ideal from user perspective, but architecturally, could be very problematic.

So I expect exposing chunk progress would make the most sense. I will update when I have some time to consider and come up with a design.

**Reply by OscarEstudilloMarin** [reply]
Thanks for the response ☺️

In production the upload does look smoother because the chunk uploads faster than the animation, however on slow connections it will still look like steps.

It is not something critical, but if you have some time and you want to add it I would be much apreciated.

Regards,
Oscar.
